schematic/manifest/generator.py

```diff
@@ -439,7 +439,7 @@ def _gather_dependency_requirements(self, json_schema, required_metadata_fields)
 
         return required_metadata_fields
 
-    def _add_root_to_component(self, required_metadata_fields):
+    def _add_root_to_component(self, required_metadata_fields:dict):
```

Comment:
One way to be more specific than `dict` is to type the keys and/or values of the dict. For example if you know the keys are always strings you can do `dict[str, Any]`, or if you knows the values are always integers you can do `dict[str, int]`

Reply:
@andrewelamb Thanks for the tip. I could add `dict[str, Any]`